### PR TITLE
FEATURE: RAIL-642 Add support for custom loading and error components in visualization, kpi and AfmComponents

### PR DESCRIPTION
**Pull request by tomasreichmann**

### CODE REVIEW

File: src/components/simple/tests/Kpi.spec.tsx [hunk 1]
```diff
@@ -17,18 +22,27 @@ class DummyExecuteEmpty extends React.Component<any, null> {
...
+
+class DummyExecuteError extends React.Component<any, null> {
+    public render() {
+        return this.props.children({ result: null, isLoading: true, error: { status: ErrorStates.BAD_REQUEST } });
+    }
+}
+
 describe('Kpi', () => {
-    function createComponent(format?: string, ExecuteComponent = DummyExecute) {
+    function createComponent({ ExecuteComponent = DummyExecute, ...otherProps}: any = {}) {
```
Comment (zbycz) [inline src/components/simple/tests/Kpi.spec.tsx]: cool

**Comment by tomasreichmann** [comment]
ok to test

**Comment by tomasreichmann** [comment]
ok to test

**Comment by langhammerp** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/539/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/446/ : SUCCESS in 52s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-merge-compliance-check/142/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/448/ : SUCCESS in 53s
- https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/143/ : SUCCESS in 4m 03s

**Pull request merged**